File: seed/php-model/trace/src/Submission/GradedResponse.php
<?php

namespace Seed\Submission;

use Seed\Core\Json\SerializableType;
use Seed\Core\Json\JsonProperty;
use Seed\Core\Types\ArrayType;

class GradedResponse extends SerializableType
{
    /**
     * @var string $submissionId
     */
    #[JsonProperty('submissionId')]
    public string $submissionId;

    /**
     * @var array<string, TestCaseResultWithStdout> $testCases
     */
    #[JsonProperty('testCases'), ArrayType(['string' => TestCaseResultWithStdout::class])]
    public array $testCases;

    /**
     * @param array{
     *   submissionId: string,
     *   testCases: array<string, TestCaseResultWithStdout>,
     * } $values
     */
    public function __construct(
        array $values,
    ) {
        $this->submissionId = $values['submissionId'];
        $this->testCases = $values['testCases'];
    }
}

File: seed/ruby-sdk-v2/trace/lib/seed/submission/types/graded_response.rb
# frozen_string_literal: true

module Seed
  module Submission
    module Types
      class GradedResponse < Internal::Types::Model
        field :submission_id, -> { String }, optional: false, nullable: false, api_name: "submissionId"
        field :test_cases, lambda {
          Internal::Types::Hash[String, Seed::Submission::Types::TestCaseResultWithStdout]
        }, optional: false, nullable: false, api_name: "testCases"
      end
    end
  end
end
